google/resource_compute_instance.go

```diff
@@ -1239,6 +1242,20 @@ func resourceComputeInstanceDelete(d *schema.ResourceData, meta interface{}) err
 	return nil
 }
 
+func resourceComputeInstanceImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.Split(d.Id(), "/")
+
+	if len(parts) != 3 {
+		return nil, fmt.Errorf("Invalid import id %q. Expecting {project}/{zone}/{instance_name}", d.Id())
```

Comment:
Do you think it would be worth it to also allow users to just import by name and then rely on the provider-level project/zone?

Reply:
I will refactor what we have done in some other resources to make it reusable and reuse it here. I wanted to keep that PR simple.